domain/lib/src/usecases/received/remove_deleted_received_share_from_local_database.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:domain/domain.dart';
+import 'package:collection/collection.dart';
+
+class RemoveDeletedReceivedShareFromLocalDatabaseInteractor {
+  final ReceivedShareRepository _receivedShareRepository;
+
+  RemoveDeletedReceivedShareFromLocalDatabaseInteractor(
+      this._receivedShareRepository);
+
+  Future<void> execute(
+      List<ReceivedShare> receivedShares, String recipient) async {
+    var localRecievedShares = await _receivedShareRepository
+        .getAllReceivedShareOfflineByRecipient(recipient);
+
+    for (final localReceived in localRecievedShares) {
+      final receivedShare = receivedShares.firstWhereOrNull(
+          (received) => received.shareId == localReceived.shareId);
+      if (receivedShare == null) {
+        await _receivedShareRepository.disableOffline(
+            localReceived.shareId, localReceived.localPath ?? '');
+      }
+    }
+  }
```

Comment:
To improve performance should use

```suggestion
 final receivedShareIds = receivedShares.map((received) => received.shareId).toSet();

  for (final local in localReceivedShares) {
    if (!receivedShareIds.contains(local.shareId)) {
      await _receivedShareRepository.disableOffline(local.shareId, local.localPath ?? '');
    }
  }
```